src/wallet/home_cards/manager.rs

```diff
@@ -0,0 +1,92 @@
+use crate::prelude::*;
+use std::sync::RwLock;
+
+#[uniffi::export(with_foreign)]
+#[async_trait::async_trait]
+pub trait HomeCardsStorage: Send + Sync {
+    async fn save_cards(&self, encoded_cards: BagOfBytes) -> Result<()>;
+
+    async fn load_cards(&self) -> Result<Option<BagOfBytes>>;
+}
+
+#[uniffi::export(with_foreign)]
+#[async_trait::async_trait]
+pub trait HomeCardsObserver: Send + Sync + std::fmt::Debug {
+    fn handle_cards_update(&self, cards: HomeCards);
+}
+
+#[derive(uniffi::Object)]
+pub struct HomeCardsManager {
+    #[allow(dead_code)] // Remove this line after gateway_client is being used
+    parser: DeferredDeepLinkParser,
+    #[allow(dead_code)] // Remove this line after cards_storage is being used
+    cards_storage: Arc<dyn HomeCardsStorage>,
+    observer: Arc<dyn HomeCardsObserver>,
+    cards: RwLock<HomeCards>,
+}
+
+impl HomeCardsManager {
```

Comment:
This logic is being worked on by @sergiupuhalschi-rdx , as part of [ABW-3466](https://radixdlt.atlassian.net/browse/ABW-3466)